ledger/ledger-api-client/src/main/scala/com/digitalasset/ledger/client/services/commands/tracker/CompletionResponse.scala

```diff
@@ -92,22 +93,14 @@ object CompletionResponse {
   private[daml] def toException(response: TrackedCompletionFailure): StatusException =
     response match {
       case QueueCompletionFailure(failure) =>
-        toException(failure)
+        val metadata = failure.metadata
+        val statusBuilder = extractStatus(failure)
+        buildException(metadata, statusBuilder)
       case QueueSubmitFailure(status) =>
-        val protoStatus =
-          rpc.Status
-            .newBuilder()
-            .setCode(status.getCode.value())
-            .setMessage(Option(status.getDescription).getOrElse("Failed to submit request"))
-        buildException(Map.empty[String, String], protoStatus)
+        val statusBuilder = GrpcStatus.toJavaBuilder(status)
+        buildException(Map.empty[String, String], statusBuilder)
     }
 
-  def toException(response: CompletionResponse.CompletionFailure): StatusException = {
-    val metadata = response.metadata
-    val status = extractStatus(response)
-    buildException(metadata, status)
-  }
-
   private def buildException(metadata: Map[String, String], status: rpc.Status.Builder) = {
     val errorInfo = rpc.ErrorInfo.newBuilder().putAllMetadata(metadata.asJava).build()
     val details = Any.pack(errorInfo)
```

Comment:
I think you also need to update this so that if there's already an `ErrorInfo`, it modifies it, rather than adding a new one.